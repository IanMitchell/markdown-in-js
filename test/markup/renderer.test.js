import React from 'react'
import renderer from 'react-test-renderer'
import markdown from '../../lib'

const App = () => markdown`
# title

This is some text <span style=${{ fontWeight: 'bold' }}> we _here_ </span>

This is more text. And some more. And more.
\`\`\`jsx
let x = alpha
function xyz(){

}
\`\`\`

here's some \`inline code\`

${'I interpolated some text here'}

and <span style=${{ fontWeight: 'bold' }} className=${'someclasssomething else'}>
this is some inline _italicized_ *bold* html
and this bit is _${'interpolated'}_
</span>

and <span> what about _this_ </span>

This is an H1
=============

This is an H2
-------------

# This is an H1

## This is an H2

###### This is an H6
`

describe('markdown-in-js', () => {
  test('Markdown correctly compiles', () => {
    const component = renderer.create(<App />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
