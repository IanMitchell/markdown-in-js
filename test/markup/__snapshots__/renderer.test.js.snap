// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`markdown-in-js Markdown correctly compiles 1`] = `
<div
  className="_markdown_"
>
  <h1>
    title
  </h1>
  <p>
    This is some text 
    <span
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
       we 
      <em>
        here
      </em>
       
    </span>
  </p>
  <p>
    This is more text. And some more. And more.
  </p>
  <pre>
    <code
      className="language-jsx"
    >
      let x = alpha
      function xyz(){
      
      }
      
    </code>
  </pre>
  <p>
    here's some 
    <code>
      inline code
    </code>
  </p>
  <p>
    I interpolated some text here
  </p>
  <p>
    and 
    <span
      className="someclasssomething else"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      this is some inline 
      <em>
        italicized
      </em>
       
      <em>
        bold
      </em>
       html and this bit is 
      <em>
        interpolated
      </em>
    </span>
  </p>
  <p>
    and 
    <span>
       what about 
      <em>
        this
      </em>
       
    </span>
  </p>
  <h1>
    This is an H1
  </h1>
  <h2>
    This is an H2
  </h2>
  <h1>
    This is an H1
  </h1>
  <h2>
    This is an H2
  </h2>
  <h6>
    This is an H6
  </h6>
</div>
`;
